src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -110,23 +116,36 @@ TableOfContentsInputItem GetItem(HeadingBlock block)
                 }
 
                 var link = block.Inline.FirstOrDefault(l => l is LinkInline);
-                var literalLink = block.Inline.FirstOrDefault(l => l is LiteralInline);
                 if (link != null && link is LinkInline linkInline)
                 {
                     if (!string.IsNullOrEmpty(linkInline.Url))
                         currentItem.Href = linkInline.Url;
                     if (!string.IsNullOrEmpty(linkInline.Title))
                         currentItem.DisplayName = linkInline.Title;
 
-                    literalLink = linkInline.FirstOrDefault(l => l is LiteralInline);
+                    currentItem.Name = GetName(linkInline.Select(l => l as LeafInline));
                 }
 
-                if (literalLink != null && literalLink is LiteralInline literalInline)
-                {
-                    currentItem.Name = literalInline.Content.ToString();
-                }
+                currentItem.Name = currentItem.Name ?? GetName(block.Inline.Select(l => l as LeafInline));
 
                 return currentItem;
+
+                string GetName(IEnumerable<LeafInline> literalInlines)
+                {
+                    literalInlines = literalInlines?.Where(l => l != null && !l.Span.IsEmpty);
+                    if (literalInlines == null || !literalInlines.Any())
+                    {
+                        return null;
+                    }
+
+                    var start = literalInlines.First().Span.Start;
+                    var lenght = literalInlines.Last().Span.End - start + 1;
+
+                    if (start >= 0 && lenght > 0)
```

Comment:
when `start < 0` or `length < 0`, it means there is a bug somewhere in markdig or our code, shall we let docfx crash?

Reply:
crash is not really user friendly, let's add Debug.Assert here?